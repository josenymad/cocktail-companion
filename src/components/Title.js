import React from "react";
import "../styles/title.css";

const Title = () => {
  return (
    <div className="title">
      <div className="navbar-titles" id="outer-container">
        <div id="page-wrap">
          <h1 className="navbar-title__header">Cocktail Companion</h1>
          <h2 className="navbar-title__slogan">The Ultimate Cocktail Guide</h2>
        </div>
      </div>
    </div>
  );
};

export default Title;
