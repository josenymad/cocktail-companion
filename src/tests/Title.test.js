import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Title from "../components/Title";

describe("Title", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Title />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders cocktail companion", () => {
    render(
      <MemoryRouter>
        <Title />
      </MemoryRouter>,
    );
    const title = screen.getByText(/cocktail companion/i);
    expect(title).toBeInTheDocument();
  });

  it("renders the ultimate cocktail guide", () => {
    render(
      <MemoryRouter>
        <Title />
      </MemoryRouter>,
    );
    const slogan = screen.getByText(/the ultimate cocktail guide/i);
    expect(slogan).toBeInTheDocument();
  });
});
