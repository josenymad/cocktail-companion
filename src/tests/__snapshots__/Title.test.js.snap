// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title renders correctly 1`] = `
<DocumentFragment>
  <div
    class="title"
  >
    <div
      class="navbar-titles"
      id="outer-container"
    >
      <div
        id="page-wrap"
      >
        <h1
          class="navbar-title__header"
        >
          Cocktail Companion
        </h1>
        <h2
          class="navbar-title__slogan"
        >
          The Ultimate Cocktail Guide
        </h2>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
